src/controllers/__tests__/authenticationController.test.js

```diff
@@ -0,0 +1,314 @@
+yOvnpZJZWyhjmiRqkBW7jkZHcmFRceeoVkXGn4bAFGGcqESFMVmaigTEm1c6gJpR
+ojo75smqyWxngEk1XLctn1+Qhb5SCbd2oHh0oLE5jpHyrxfxw8V+N2Hty26GavJE
+7i9jORbjeQCMkbggt0FahmlmaZr20akK8wNGMHDcpnMslJPxHl6uKxjAfe6sbNqj
+Wxfcnirm05Jh5gYNT4vkwC1vx6AZpS2G9pxOV1q5GapuvUBqwNu+EH1ufMRRXvu0
++GtJ4WtsErOakSF4KMezrMqKCrVPoK5SGxQMD/kwEQ8HfUPpim3cdi3RVmqQjsi/
+on6DMn/xTQIDAQABAoIBAWo61Yw8Q/m9CwrgPyPRQm2HBwx/9/MPbaovSdzTrIm6
+Gmg7yDYVm/kETj3JQ/drUzKIbj6t9LXvUizOUa2VSMJ0yZTYsnDHuywi8nf0uhgO
+5pAca0aJLJ792hEByOx+EeUSN3C3i35vfbn8gwYoAHjrVA8mJrAEsawRbdVpNj6j
+IWAKTmsZK0YLdcNzWshSYW9wkJNykeXHUgKk2YzGUIacMgC+fF3v3xL82xk+eLez
+dP5wlrzkPz8JKHMIomF5j/VLuggSZx0XdQRvZrkeQUbJqRy2iXa43B+OlEiNvd2Q
+0AiXur/MhvID+Ni/hBIoeIDyvvoBoiCTZbVvyRnBds8BAoGBDIfqHTTcfXlzfipt
+4+idRuuzzhIXzQOcB0+8ywqmgtE4g9EykC7fEcynOavv08rOSNSbYhjLX24xUSrd
+19lckZIvH5U9nJxnyfwrMGrorCA2uPtck8N9sTB5UWif31w/eDVMv30jRUyMel7l
+tp96zxcPThT1O3N4zM2Otk5q2DvFAoGBDBngF4G9dJ5a511dyYco2agMJTvJLIqn
+kKw24KOTqZ5BZKyIea4yCy9cRmNN84ccOy3jBuzSFLNJMYqdDCzH46u0I4anft83
+aqnVa4jOwjZXoV9JCdFh3zKJUgPU4CW0MaTb30n3U4BAOgkHzRFt55tGT6xRU1J+
+jX5s03BFfQ/pAoGBCsRqtUfrweEvDRT2MeR56Cu153cCfoYAdwPcDHeNVlDih9mk
+4eF0ib3ZXyPPQqQ8FrahAWyeq9Rqif0UfFloQiljVncNZtm6EQQeNE9YuDZB7zcF
+eG59PViSlhIZdXq1itv5o3yqZux8tNV/+ykUBIgi/YvioH/7J7flTd8Zzc2lAoGB
+CqdVNRzSETPBUGRQx7Yo7scWOkmSaZaAw8v6XHdm7zQW2m1Tkd0czeAaWxXecQKI
+hkl10Ij6w6K8U9N3RFrAeN6YL5bDK92VSmDPNmcxsKZrK/VZtj0S74/sebpJ1jUb
+mYFM2h6ikm8dHHsK1S39FqULl+VbjAHazPN7GAOGCf7RAoGBAc0J9j+MHYm4M18K
+AW2UB26qvdc8PSXE6i4YQAsg2RBgtf6u4jVAQ8a8OA4vnGG9VIrR0bD/hFTfczg/
+ZbWGZ+42VH2eDGouiadR4rWzJNQKjWd98Y5PzxdEjd6nneJ68iNqGbKOT6jXu8qj
+nCnxP/vK5rgVHU3nQfq+e/B6FVWZ
+-----END PRIVATE KEY-----
+`;
+const time = 1518010929530;
+const user = {
+  created_at: time,
+  family_name: "Lusso",
+  fiscal_code: "XUZTCT88A51Y311X",
+  name: "Luca",
+  nameID: "lussoluca",
+  nameIDFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+  preferred_email: "luca.lusso@wellnet.it",
+  sessionIndex: "123",
+  spid_idp: "xxx",
+  token: "123"
+};
+const validUserPayload = {
+  email: "luca.lusso@wellnet.it",
+  familyName: "Lusso",
+  fiscalNumber: "XUZTCT88A51Y311X",
+  issuer: {
+    _: "xxx"
+  },
+  name: "Luca",
+  nameID: "lussoluca",
+  nameIDFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+  sessionIndex: "123"
+};
+// invalidUser lacks the required email field.
+const invalidUserPayload = {
+  familyName: "Lusso",
+  fiscalNumber: "XUZTCT88A51Y311X",
+  issuer: {
+    _: "xxx"
+  },
+  name: "Luca",
+  nameID: "lussoluca",
+  nameIDFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+  sessionIndex: "123"
+};
+
+const mockSet = jest.fn();
+const mockDel = jest.fn();
+jest.mock("../../services/redisSessionStorage", () => {
+  return jest.fn().mockImplementation(() => {
+    return { set: mockSet, del: mockDel };
+  });
+});
+const spidStrategy_ = spidStrategy(samlKey);
+spidStrategy_.logout = jest.fn();
+
+describe("Authentication Controller acs method", () => {
```

Comment:
in the rest of the project we use the convention of naming the suites with `classname#methodname`, so this would be `AuthenticationController#acs` - sorry I should have shared this info with you in advance